Tools/Scripts/pgo-profile

```diff
@@ -0,0 +1,142 @@
+    return parser
+
+
+def merge(args):
+    path = assert_directory(args.directory)
+
+    for lib in ["JavaScriptCore", "WebCore", "WebKit"]:
+        print("Merging", lib)
+        print(shell("xcrun -sdk macosx.internal llvm-profdata merge \"{}\"/{}*.profraw -output=\"{}/{}.profdata\"".format(path, lib, path, lib)))
+
+
+def combine_parser(subparsers):
+    parser = subparsers.add_parser("combine", help="Combine directories containing *.profdata files from different platforms together.")
+    parser.add_argument("--speedometer", default=None, help="Path to the directory containing the *.profdata files from a speedometer run.")
+    parser.add_argument("--jetstream", default=None, help="Path to the directory containing the *.profdata files from a jetstream run.")
+    parser.add_argument("--output", help="Path to the directory where the output will be placed.")
+    parser.set_defaults(func=combine)
+    return parser
+
+
+def combine(args):
+    out = assert_directory(args.output)
+    args = vars(args)
+    groups = [("speedometer", 0.6), ("jetstream", 0.4)]
+    group_paths = []
+
+    for i in range(0, len(groups)):
+        group, weight = groups[i]
+        if args[group]:
+            path = assert_directory(args[group])
+            group_paths.append((path, weight))
+    
+    if len(group_paths) == 0:
+        print("Must specify at least one group.")
+        exit()
+    
+    sum = 0
+    max = 0
+    # We need to turn percentages into weights > 1, but we don't want crazy high multipliers.
+    # For example, if we have weights 0.35 and 0.65, we don't need a 7:13 ratio when 5:9 is good enough.
+    max_multiplier = 15
+    for group, weight in group_paths:
+        sum = sum + weight
+        if weight > max:
+            max = weight
+
+    gcd = int(max * max_multiplier)
+    for group, weight in group_paths:
+        gcd = math.gcd(gcd, int((weight / sum) * max_multiplier))
+    for i in range(0, len(group_paths)):
+        group, weight = group_paths[i]
+        group_paths[i] = (group, int((weight / sum) * max_multiplier) // gcd)
+    
+    print("Simplified group weights: ", group_paths)
+
+    for lib in ["JavaScriptCore", "WebCore", "WebKit"]:
+        print("Merging", lib)
+        group_input = ["--weighted-input={},\"{}/{}.profdata\"".format(weight, path, lib) for path, weight in group_paths]
+        group_input = " ".join(group_input)
+        print(shell("xcrun -sdk macosx.internal llvm-profdata merge {} -output=\"{}/{}.profdata\"".format(group_input, out, lib)))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(prog='pgo-profile')
+    subparsers = parser.add_subparsers(help='valid sub-commands', required=True, dest='sub command')
+    merge_parser(subparsers)
+    summarize_parser(subparsers)
+    combine_parser(subparsers)
+
+    args = parser.parse_args()
+    try:
+        args.func(args)
+    except subprocess.CalledProcessError as e:
+        print(e.stdout)
```

Comment:
I think `print(e)`? I not sure if `subprocess.CalledProcessError` has `e.stdout`

Reply:
It gets printed by default since it is unhandled. In this case, I tested it and it gives me the stderr from the subprocess to aid with debugging.

Reply:
OK, it looks like `e.stdout` is added since python3.5. https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError